ReactAndroid/src/main/java/com/facebook/react/views/checkbox/ReactCheckBoxManager.java

```diff
@@ -60,4 +67,44 @@ public void setOn(ReactCheckBox view, boolean on) {
     view.setOn(on);
     view.setOnCheckedChangeListener(ON_CHECKED_CHANGE_LISTENER);
   }
+
+  private static int getThemeColor(final Context context, String colorId) {
+    final TypedValue value = new TypedValue();
+    context.getTheme().resolveAttribute(getIdentifier(context, colorId), value, true);
+    return value.data;
+  }
+
+  /**
+   * The appcompat-v7 BUCK dep is listed as a provided_dep, which complains that
+   * com.facebook.react.R doesn't exist. Since the attributes are provided from a parent, we can access
+   * those attributes dynamically.
+   */
+  private static int getIdentifier(Context context, String name) {
+    return context.getResources().getIdentifier(name, "attr", context.getPackageName());
+  }
+
+  @ReactProp(name = "tintColors")
+  public void setTintColors(ReactCheckBox view, @Nullable ReadableMap colorsMap) {
+    String defaultColorIdOfCheckedState = "colorAccent";
+    int trueColor = colorsMap == null || !colorsMap.hasKey("true") ?
+            getThemeColor(view.getContext(), defaultColorIdOfCheckedState) : colorsMap.getInt("true");
```

Comment:
so, while this works, it means not specifying the `tintColors` prop causes us to call `getThemeColor` which is [discouraged](https://developer.android.com/reference/android/content/res/Resources#getIdentifier(java.lang.String,%20java.lang.String,%20java.lang.String)) so, question - do we wanna cache the color? (seems like that approach should work as activity needs to be recreated when theme is changed [SO](https://stackoverflow.com/questions/2482848/how-to-change-current-theme-at-runtime-in-android)). We could also keep the colors as ivars (I wanted to avoid it but ¯\_(ツ)_/)

Reply:
Is there no way we can make it completely optional, ie. if `tintColor` is not passed, we keep the current behavior. If `tintColor` is passed, we use the new behavior?

Reply:
When `tintColors` is undefined the code in `setTintColors` will not executed.

But say the user passes some `tintColors` to customize them, and then passes `tintColors=undefined` they would think it will switch the colors to default, but it will keep the colors previously set. That's why I explicitly set the prop to `null` in the JS side, and then the Java code receives the `null`. It's a corner case, but this should work imho.

Reply:
I agree null and undefined should be treated the same but could we just smooth over this on the JS side and send "null" to native and reset in that case?

Reply:
from JS I always send the object or null, I never send undefined: https://github.com/facebook/react-native/pull/18300/files#diff-7f3d990a8a1579593beee654dfc118feR162

so yes, null always resets, the thing it resets even when not needed (as outlined in the first comment).

Reply:
Ah I see what you mean now. As said, could we manage this in JS using local state? It's not pretty but we could make it so that tintColor is not sent if it's not set (or a special token is sent), but once it is set we use local state to indicate that it should get reverted.

This means that it would keep working the same for all existing checkboxes, and have the new and expected behavior when `tintColor` is being used.

Reply:
@cpojer it turns out I was wrong, so here's how it works:

sending `undefined` as `tintColors` from JS on first render does **not** invoke `setTintColors` in Java at all. If `undefined` is passed also in the following, JS checkbox re-renders, the Java code is **not** invoked.

sending `null` as `tintColors` from JS on first render does(!) invoke `setTintColors` in Java with `colorsMap == null` . If `null` is passed also in the following, JS checkbox re-renders, the Java code is **not** invoked. Also, when `tintColors` prop is changed from a valid colors object to undefined, the Java code will receive a `colorsMap` that is `null`. Related comment from `ReactStylesDiffMap.java`:

>  * Instances of this class are used to update {@link View} or {@link CSSNode} style properties.
>  * Since properties are generated by React framework based on what has been updated each value
>  * in this map should either be interpreted as a new value set for a style property or as a "reset
>  * this property to default" command in case when value is null (this is a way React communicates
>  * change in which the style key that was previously present in a map has been removed).

React apparently does some sort of a magic in the background, and is able to figure out on its own whether or not a native prop changed. So the way I see this is that this just works even without keeping any state around because it already is kept somewhere by React.

<img width="969" alt="screenshot 2019-02-06 at 00 18 16" src="https://user-images.githubusercontent.com/1566403/52312616-89d38d00-29ab-11e9-88af-891c265b8fdb.png">